pkg/client/block/client.go

```diff
@@ -0,0 +1,181 @@
+package block
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	"cosmossdk.io/depinject"
+	cometTypes "github.com/cometbft/cometbft/types"
+
+	"pocket/pkg/client"
+	"pocket/pkg/observable"
+	"pocket/pkg/observable/channel"
+)
+
+var (
+	_              client.BlockClient = &blockClient{}
```

Comment:
```suggestion
	_              client.BlockClient = (*blockClient)(nil)
```
This is the cosmos-sdk and go standard